### PR TITLE
gre & vti: use alternate way to check if kernel support is enabled

### PR DESCRIPTION
**Pull request by alinnastac**
When necessary support is built in kernel, gre and vti protocol
support is not enabled in netifd.

Signed-off-by: Alin Nastac <alin.nastac@gmail.com>

Thanks for your contribution to OpenWrt!

To help keep the codebase consistent and readable,
and to help people review your contribution,
we ask you to follow the rules you find in the wiki at this link
https://openwrt.org/submitting-patches

Please remove this message before posting the pull request.

### CODE REVIEW

**Comment by obsy** [comment]
Use ||, not -o. See https://github.com/openwrt/openwrt/commit/1c0e13db43b41a1d527428b82bf88a032a916e84

**Comment by alinnastac** [comment]
@obsy I fail to see the rationale behind shellcheck recommendation. AFAIK -o and -a are portable test operators (see man test).

Besides, using the syntax you propose will spawn 2 test processes instead of one.

**Comment by obsy** [comment]
All scripts use this. So please use the same syntax as the rest.

**Comment by alinnastac** [comment]
According to https://github.com/koalaman/shellcheck/wiki/SC2166, this recommendation was issued for generic test cases (e.g. test $1 -o $2), where left or right expressions might contain parenthesis.

In this particular case, none of the -o operands contain any parenthesis... But have it your way, I've pushed the variant you desired.

**Comment by dedeckeh** [comment]
@alinnastac Please split up the gre and vti changes in separate commits and increase the PKG_RELEASE as well for each package

File: package/network/config/gre/files/gre.sh [hunk 1]
```diff
@@ -291,8 +291,6 @@ proto_grev6tap_init_config() {
 }
 
 [ -n "$INCLUDE_ONLY" ] || {
-	[ -f /lib/modules/$(uname -r)/gre.ko ] && add_protocol gre
-	[ -f /lib/modules/$(uname -r)/gre.ko ] && add_protocol gretap
-	[ -f /lib/modules/$(uname -r)/ip6_gre.ko ] && add_protocol grev6
-	[ -f /lib/modules/$(uname -r)/ip6_gre.ko ] && add_protocol grev6tap
+	[ -f /lib/modules/$(uname -r)/gre.ko ] || [ -d /sys/module/ip_gre ] && { add_protocol gre; add_protocol gretap; }
+	[ -f /lib/modules/$(uname -r)/ip6_gre.ko ] || [ -d /sys/module/ip6_gre ] && { add_protocol grev6; add_protocol grev6tap; }
```
Comment (rsalvaterra) [inline package/network/config/gre/files/gre.sh:295]: I'd argue checking for the existence of the module files is completely wrong (think monolithic builds) and shouldn't even be done…

**Comment by dedeckeh** [comment]
Patches pushed to master; thx

**Pull request closed without merging**